lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
Oh, by the way! This actually came up for me real-world style a few weeks ago, but I forgot until @kevintyll brought it up.

I was writing a jsonapi client in ruby, using the [jsonapi gem](https://github.com/beauby/jsonapi) AMS also uses. That gem was blowing up on parsing due to exactly this issue.

Not to say that lib is a golden source of truth, but FYI. We gotta choose to follow the docs or the schema, that's an example of other libraries we need to play nicely with favoring the docs.